fvm/evm/emulator/emulator.go

```diff
@@ -306,6 +311,13 @@ func (bl *BlockView) DryRunTransaction(
 		txResult.GasConsumed += txResult.GasRefund
 	}
 
+	// call tracer on tx end
```

Comment:
I am not sure if this should be before or after adjusting the `txResult.GasConsumed` with the various buffers.